### DISCUSSION TITLE
What does it really do?

### DISCUSSION THREAD

**Discussion by U-Go-7** [MathRobin/Never-Consent#93] [Q&A] [answered]
Hi, my interpretation is that this addon automatically select "only essential cookies" in the supported platforms.
Is this correct?
What about  "legitimate interest"? Does it object that too?

And what about Google Consent?I expected it to disappear but it still pops up.
Thanks

**Comment by MathRobin** [comment, marked as answer]
Hi ! You're true, it **tries** to automatically select "only essentials cookies" when option is available. And that's all. No way to customize. For this, I suggest you to prefer extension like consent-o-matic
Chrome: https://chrome.google.com/webstore/detail/consent-o-matic/mdjildafknihdffpkfmmpnpoiajfjnjd
Firefox: https://addons.mozilla.org/fr/firefox/addon/consent-o-matic/

It's an opinionated choice on my own to just reject any other usage of cookies.

Concerning Google Consent, it's still in development, but as i develop this extension on my free time and alone, and as a father of two young daemons, I don't have as time as wanted to keep tracking of platform consents changes and to include all platforms i could want. Sadly. Don't hesitate to ask any other questions, you're welcome!

**Reply by U-Go-7** [reply]
Well I like Demons and Daemons, and at least it seems they did not manage to suppress your good mood.
Yet :)

I understand that I can't customize which cookies to allow and which not.
But, what happens to the legitimate interest thing? Does your addon object them all or does it only work on the normal cookies?

**Reply by MathRobin** [reply]
I just target only essential cookies, the rest is fully rejected without any discussions. Sadly i just develop this extension to polyfill the non-respect of "do not track" flag of my browser settings. I can accept them all too as i use ublock and other blocker things, but it's a matter of principle, i don't want any non-stricly useful cookies on my machine. Even if it's not the best way anymore to identity somebody

**Reply by U-Go-7** [reply]
Well, if your addon rejects all legitimate interest I really do not need the o-matic one. Why did you suggest it? Maybe you thought I did not want to reject everything? (I absolutely want!)
Anyway, in most consent forms, even where "only essential cookies" is selected as default, the legitimate interest stuff is also selected. It's like if for law they are not considered real cookies but something else which can be allowed (for "legitimate interest") even when one wants only the essential stuff.
As far as I read, also o-matic doesn't handle legitimate interest: [https://github.com/cavi-au/Consent-O-Matic/issues/31](url)
You sure that your addon reject legitimate interest? Like, 100% sure?

About the rest, I totally get it, I feel the same.
You mention Ublock, I use that too, although I at times prefer Adguard (seems to block more stuff). And I just was about to ask this question somewhere, but maybe you know already: do these addons reject all not essential cookies too? If they do, why do we need yours? If they do not, what happen when they manage to make the consent popup disappear? Do they reject or just hide and leave the default choice as it is (which mostly is with all legitimate interest accepted)?

**Reply by MathRobin** [reply]
Will never be 100% sure. Some systems are just like a facade, they just say "ok you don't want, oh look to this very long time to save your preferences, please click here to accept everything and it will be quicker" (lol seriously ?! saving nothing is more longer than saving a lot of shit ?! do they think we're stupid ?). Some websites just doesn't care too. They provide a tool to manage cookies but they already have installed everything, whatever you say to platform. Sadly for this kind of cases, i can't do anything (except ublock, ...). And to be honest, google, facebook & co don't need anymore to use cookies, they are so deeply integrated in our lives that just cookies are nothing for them. I try to not use google, but as a web dev, i use chrome for my work, i have an android phone (the other alternative is iphone, well, just the same shit but just more expensive), i use whatsapp because most of my contacts don't want to install signal in addition of messages, kakaotalk (part of my family is korean, so they mostly use this one) and other another-messaging-app.

To answer your first question, this extension tries to find the "reject all" button, the "continue without accepting" (or "continuer sans accepter"). I just want nothing of your business on my machine. Lot of websites now provides a paid alternative when you don't want cookies. I think it's ok if they don't track me. I like to have the choice, still accepting that people are paid for their work. Just want to be able to don't sell my data and my as* to anybody.

When my first child was born (will not say to internet if it's a girl or a boy), it tooks 2 days to receive its first ad by paper in our letterbox at its complete name, hell, it was still at maternity ward. Don't want this for my child. This is a sign that they already have too much data about me, my life and my family.
